lib/resty/openidc.lua

```diff
@@ -576,6 +578,13 @@ function openidc.authenticate(opts, target_url)
     return openidc_authorize(opts, session, target_url), session
   end
 
+  -- silently reauthenticate if necessary (mainly used for session refresh/getting updated id_token data)
+  if opts.refresh_session_internal and (session.data.last_authenticated == nil or (session.data.last_authenticated+reauth_delay) < ngx.time()) then
```

Comment:
Seems that `opts.refresh_session_internal ` and `reauth_delay` are used without being defined? An alternative may be to define an optional custom callback function in `opts` that implements the requested behavior.